include/ScratchViews.h

```diff
@@ -223,32 +248,31 @@ void fill_pre_req_data(
   ScratchViews& prereqData)
 {
   int nodesPerElem = topo.num_nodes();
-  const stk::mesh::Entity* nodes = nullptr;
 
   // extract master elements
   MasterElement *meSCS = dataNeeded.get_cvfem_surface_me();
   MasterElement *meSCV = dataNeeded.get_cvfem_volume_me();
 
   const FieldSet& neededFields = dataNeeded.get_fields();
-  for(const stk::mesh::FieldBase* fieldPtr : neededFields) {
-    stk::mesh::EntityRank fieldEntityRank = fieldPtr->entity_rank();
+  for(const FieldInfo& fieldInfo : neededFields) {
+    stk::mesh::EntityRank fieldEntityRank = fieldInfo.field->entity_rank();
+    unsigned scalarsPerEntity = fieldInfo.scalarsPerEntity;
     if (fieldEntityRank==stk::topology::ELEM_RANK) {
```

Comment:
Okay, so this means that there is still an interface to elemental data that is shared?